File: 1468-check-if-n-and-its-double-exist/1468-check-if-n-and-its-double-exist.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var checkIfExist = function(arr) {
    const seen =  new Map();
    arr.forEach((num,index)=> seen.set(num , index));
    var answer = false;
    arr.forEach((num,index)=> {
        if(seen.has(arr[index] * 2) && seen.get(arr[index] * 2) != index)
                answer = true;
    });
    return answer;
}

File: 1468-check-if-n-and-its-double-exist/1468-check-if-n-and-its-double-exist.java
class Solution {
    public boolean checkIfExist(int[] arr) {
        HashMap<Integer , Integer > map = new HashMap<>();
        var i = 0;
        for( int num : arr)
           map.put(num , i++);
        for( i = 0; i < arr.length; i++){
            if( map.containsKey(arr[i]*2) && map.get(arr[i]*2) != i)
               return true;
        }
        return false;
    }
}